test/style/style_layer.test.cpp

```diff
@@ -268,3 +274,24 @@ TEST(Layer, Observer) {
     layer->setLineCap(lineCap);
     EXPECT_FALSE(layoutPropertyChanged);
 }
+
+TEST(Layer, DuplicateLayer) {
+    util::RunLoop loop;
+
+    //Setup style
+    StubFileSource fileSource;
+    Style style { fileSource, 1.0 };
+    style.setJSON(util::read_file("test/fixtures/resources/style-unused-sources.json"));
+
+    //Add initial layer
+    style.addLayer(std::make_unique<LineLayer>("line", "unusedsource"));
+
+    //Try to add duplicate
+    try {
+        style.addLayer(std::make_unique<LineLayer>("line", "unusedsource"));
+        FAIL() << "Should not have been allowed to add a duplicate layer id";
+    } catch (std::runtime_error) {
```

Comment:
The rule for exceptions in C++ is to throw by value and catch a `const` reference